kafka/src/main/java/io/specmesh/kafka/ProvisionStatus.java

```diff
@@ -0,0 +1,252 @@
+    @NoArgsConstructor(access = AccessLevel.PRIVATE)
+    @Accessors(fluent = true)
+    @SuppressFBWarnings
+    public static class SchemaStatus {
+        private CRUD crud;
+        private int id;
+        private String schemaSubject;
+        private SchemaInfo schemaInfo;
+        private String schemaPath;
+        private Exception exception;
+    }
+
+    /** Acl Provisioning status */
+    @Builder
+    @Data
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    @NoArgsConstructor(access = AccessLevel.PRIVATE)
+    @Accessors(fluent = true)
+    @SuppressFBWarnings
+    public static class Acls {
+        @Builder.Default private Set<AclBinding> aclsToCreate = Collections.emptySet();
+        @Builder.Default private Set<AclBinding> aclsCreated = Collections.emptySet();
+        private Map<String, AclStatus> status;
+        private Exception exception;
+
+        public void build() {
+            status =
+                    aclsToCreate.stream()
+                            .map(
+                                    acl -> {
+                                        final var aclStatus =
+                                                AclStatus.builder()
+                                                        .name(acl.toString())
+                                                        .entry(acl.entry())
+                                                        .pattern(acl.pattern())
+                                                        .crud(CRUD.CREATE);
+
+                                        final var wasCreated =
+                                                aclsCreated.stream()
+                                                        .filter(
+                                                                createdAcl ->
+                                                                        createdAcl
+                                                                                .toString()
+                                                                                .equals(
+                                                                                        acl
+                                                                                                .toString()))
+                                                        .findFirst();
+                                        if (wasCreated.isPresent()) {
+                                            aclStatus.crud(CRUD.CREATED);
+                                        }
+                                        if (exception != null) {
+                                            aclStatus.crud(CRUD.FAILED);
+                                            aclStatus.exception(exception);
+                                        }
+                                        return aclStatus.build();
+                                    })
+                            .collect(Collectors.toMap(AclStatus::name, aclStatus -> aclStatus));
```

Comment:
Yuk - not reviewing

Reply:
ill do the builder stuff and sort this in the next PR